tests/lava/proc/sdn/test_models.py

```diff
@@ -0,0 +1,265 @@
+        """Tests fixed point sigma delta reconstruction with cumulative error.
+        The max absolute error must be smaller than threshold.
+        """
+        num_steps = 100
+        wgt_exp = 6
+        state_exp = 6
+        vth = 10 << (wgt_exp + state_exp)
+        input, output = self.run_test(
+            num_steps=num_steps,
+            vth=vth,
+            act_mode=ACTIVATION_MODE.Unit,
+            wgt_exp=wgt_exp,
+            state_exp=state_exp,
+            cum_error=True,
+        )
+
+        error = np.abs(input - output).max()
+
+        if verbose:
+            print(f'Max abs error = {error}')
+        self.assertTrue(error < vth * (1 << wgt_exp))
+
+    def test_reconstruction_cum_error_float(self) -> None:
+        """Tests floating point sigma delta reconstruction with cumulative
+        error. The max absolute error must be smaller than threshold.
+        """
+        num_steps = 100
+        wgt_exp = 0
+        state_exp = 0
+        vth = 10
+        input, output = self.run_test(
+            num_steps=num_steps,
+            vth=vth,
+            act_mode=ACTIVATION_MODE.Unit,
+            wgt_exp=wgt_exp,
+            state_exp=state_exp,
+            cum_error=True,
+            tag='floating_pt'
+        )
+
+        error = np.abs(input - output).max()
+
+        if verbose:
+            print(f'Max abs error = {error}')
+        self.assertTrue(error < vth * (1 << wgt_exp))
+
+    def test_reconstruction_relu_fixed(self) -> None:
+        """Tests fixed point sigma delta reconstruction with ReLU.
+        The max absolute error must be smaller than threshold.
+        """
+        num_steps = 100
+        wgt_exp = 0
+        state_exp = 0
+        vth = 10 << (wgt_exp + state_exp)
+        input, output = self.run_test(
+            num_steps=num_steps,
+            vth=vth,
+            act_mode=ACTIVATION_MODE.ReLU,
+            wgt_exp=wgt_exp,
+            state_exp=state_exp,
+            cum_error=False,
+        )
+
+        error = np.abs(np.maximum(input, 0) - output).max()
+
+        if verbose:
+            print(f'Max abs error = {error}')
+        self.assertTrue(error < vth * (1 << wgt_exp))
+
+    def test_reconstruction_relu_float(self) -> None:
+        """Tests floating point sigma delta reconstruction with ReLU.
+        The max absolute error must be smaller than threshold.
+        """
+        num_steps = 100
+        vth = 10
+        wgt_exp = 0
+        state_exp = 0
+        input, output = self.run_test(
+            num_steps=num_steps,
+            vth=vth,
+            act_mode=ACTIVATION_MODE.ReLU,
+            wgt_exp=wgt_exp,
+            state_exp=state_exp,
+            cum_error=False,
+            tag='floating_pt',
+        )
+
+        error = np.abs(np.maximum(input, 0) - output).max()
+
+        if verbose:
+            print(f'Max abs error = {error}')
+        self.assertTrue(error < vth * (1 << wgt_exp))
+
+
+if __name__ == '__main__':
```

Comment:
I don't think we are doing `if __name__ == '__main__'` in all unit-tests. I am guessing you are doing this to run tests through command line?

Reply:
Yeah. I'll remove it.